include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,125 @@ typedef enum
  */
 extern DECLSPEC char *SDLCALL SDL_GetUserFolder(SDL_Folder folder);
 
+
+/* Abstract filesystem interface */
+
+typedef enum SDL_StatPathType
+{
+    SDL_STATPATHTYPE_FILE, /**< a normal file */
+    SDL_STATPATHTYPE_DIRECTORY, /**< a directory */
+    SDL_STATPATHTYPE_OTHER /**< something completely different like a device node (not a symlink, those are always followed) */
+} SDL_StatPathType;
+
+typedef struct SDL_Stat
+{
+    SDL_StatPathType filetype;  /* is this a file? a dir? something else? */
+    Sint64 filesize;    /* size in bytes */
+    Sint64 modtime;     /* SDL filesystem timestamp */
```

Comment:
What are the units for these, if I want to write my own filesystem implementation?

Reply:
Let's document it, since it's part of the ABI.

Reply:
Actually, let's define it as NS since the epoch, so we're consistent with other timestamp sources in SDL and future proof for higher resolution timestamps in the future.

Reply:
FYI, nanosecond resolution means a 64-bit signed overflow in 2261AD. For unsigned you need to add around 661 years. Should be safe?

Reply:
If it's not safe, it'll be someone else's problem!  :)

Reply:
> NS since the epoch

Do we want the Unix epoch? If we're not doing seconds, there's no reason to specifically choose that one. [There's lots of good options to choose from.](https://en.wikipedia.org/wiki/Epoch_(computing)#Notable_epoch_dates_in_computing)

Reply:
Sure, it'll make converting from POSIX file times easy.

Reply:
4ed69b84433232ceebb764f38d80f9fded91cfbe fixes the timestamps